### PR TITLE
Patch highlights format to List[Dict]

### PR DESCRIPTION
**Pull request by wanliAlex**
Patch highlights format to List[Dict]

### CODE REVIEW

**Review by pandu-k — DISMISSED** [review]

File: src/marqo/tensor_search/tensor_search.py [hunk 1]
```diff
@@ -1312,7 +1312,7 @@ def _lexical_search(
     # Set the _highlights for each doc as [] to follow Marqo-V1's convention
     if highlights:
         for docs in gathered_docs['hits']:
-            docs['_highlights'] = {}
+            docs['_highlights'] = [{}]
```
Comment (pandu-k) [inline src/marqo/tensor_search/tensor_search.py]: this needs to be an empty list - an empty dict inside an empty list means that it is no longer an empty list

File: src/marqo/s2_inference/reranking/rerank.py [hunk 2]
```diff
@@ -1,15 +1,16 @@
...
 from marqo.s2_inference.logger import get_logger
+from marqo.s2_inference.reranking.cross_encoders import ReRankerText, ReRankerOwl
+from marqo.s2_inference.reranking.enums import ResultsFields
+from marqo.s2_inference.types import Dict, List
 
 logger = get_logger(__name__)
 
 def rerank_search_results(search_result: Dict, query: str, model_name: str, device: str, 
-                searchable_attributes: List[str] = None, num_highlights: int = 1, 
+                searchable_attributes: List[str], num_highlights: int = 1,
```
Comment (farshidz) [inline src/marqo/s2_inference/reranking/rerank.py]: Why did you remove the default value here? A bit concerned about changes here as I don't believe we have test coverage?

File: src/marqo/tensor_search/tensor_search.py (hunk 1 above)
Comment (wanliAlex) [reply src/marqo/tensor_search/tensor_search.py]: Done

File: src/marqo/s2_inference/reranking/rerank.py (hunk 2 above)
Comment (wanliAlex) [reply src/marqo/s2_inference/reranking/rerank.py]: There is a value check before called this function. But I can add it back.

**Review by pandu-k — APPROVED** [review]
